src/main/java/com/internhub/data/positions/scrapers/QueuedPositionScraper.java

```diff
@@ -0,0 +1,81 @@
+package com.internhub.data.positions.scrapers;
+
+import com.google.common.collect.Lists;
+import com.internhub.data.models.Company;
+import com.internhub.data.models.Position;
+import com.internhub.data.positions.scrapers.candidates.QueuedCandidate;
+
+import java.util.*;
+
+public abstract class QueuedPositionScraper<C extends QueuedCandidate> implements PositionScraper {
+    private Queue<C> frontier;
```

Comment:
I know you want this whole frontier thing for the BFS because technically in CS classes they teach BFS like this but it's just weird to see - frontier doesn't give any info of what is inside the queue.